upa/contracts/EllipticCurveUtils.sol

```diff
@@ -62,8 +62,10 @@ library EllipticCurveUtils {
         return addmod(x, y, PRIME_Q);
     }
 
-    /// Subtracts two Fq elements.
+    /// Subtracts two Fq elements.  Note, this assumes that y is well formed
+    /// (< field modulus, e.g. with checkFq).
     function fqSub(uint256 x, uint256 y) internal pure returns (uint256) {
+        require(y < PRIME_Q);
```

Comment:
We often don't do this for the other functions, but this extra require was suggested in the audit report and there is nothing wrong with asserting that the assumption hold.  We could remove it again to be consistent with other functions, but I thought in this case it might make sense as it's the only `fq*` function that makes this assumption.